lib/QC_TreeMap.qpp

```diff
@@ -88,13 +88,31 @@ nothing TreeMap::put(string path, any value) {
    tm->put(path, value, xsink);
 }
 
-//! Retrieves a value from the TreeMap.
+//! Retrieves a value from the TreeMap and returns optionally unmatched path suffix
 /** Looks for an entry whose key is the longest prefix of \c path.
     @param path the path to lookup
+    @param unmatched path suffix, reference to a *string compatible type is needed, if \c path not found then returns NOTHING,  @since %Qore 0.8.13
```

Comment:
here it would be good to put documentation reference to the type: `*string` -> `ref @string_or_nothing_type "*string"`